### ISSUE TITLE
Problems with function-expressions.

### ISSUE THREAD

**Issue by DAGapps** [ricardofbarros/linter-js-standard#169]
When a self-invoking function-expression is on the bottom of a script, the linter throws a warning: 
'Unexpected newline between function and ( of function call.'
Because the linter does not understand that a expression bracket is standalone and assigned to the following wrapped function. Not the function or object in front. 

Example:

```javascript
var foo = {
  'value': 42
}

console.log(foo.value)

(function () {
  return
})()
```

js-standard:

```
Warning
Unexpected space between function name and paren.at line 5 col 1
Warning
Unexpected newline between function and ( of function call.at line 7 col 1
```

You can fix this by a workaround by adding a semicolon before expression function.
But this is not solution. Because extra semicolons: 

```
var foo = {
  'value': 42
}

console.log('Hello');
console.log(foo.value);

(function () {
  return
})()

```

js-standard:

```
Warning
Extra semicolon.at line 5 col 1
```

I'm sure, this is a bug. Or what's wrong?

**Comment by DAGapps** [comment]
Ok, I see now that a semicolon in front of the expression is wanted and the correct way.
But in my optinion, this looks really ugly. But, ok. This is not a bug. 
See here: http://standardjs.com/rules.html (no-unexpected-multiline)
